### ISSUE TITLE
Console message "Failed to compute"/TypeError 

### ISSUE THREAD

**Issue by woodlandhunter** [ractivejs/ractive#2092]
On edge (0.8.0), getting this message in the console for the fiddle below:

```
Ractive.js: Failed to compute @current.akey.0.astring.slice(0,4): _0 is undefined
TypeError: _0 is undefined
```

Fiddle:

http://jsfiddle.net/wax27gr9/

(Edit: I had a different fiddle attached initially but i trimmed it down a little)

**Comment by YRathod** [comment]
output of the above fiddle is same in 0.7.3 and 0.8.0 except 0.8.0 throws an error showing something wrong in data. 

theitems = [
    {akey: [{astring: '21489182'}]},
    {akey: []},
]

Here, astring is blank for the theitems[1] , hence slice is failing for the index 1. 
Error thrown here looks like valid and useful to trace the root cause. 

I think this may be not a bug but better error handing approach.

**Comment by woodlandhunter** [comment]
I believe the {{#.akey}} section shouldn't be entered for an empty list. (Same error when using a more explicit each.)

**Comment by Rich-Harris** [comment]
This looks like another manifestation of #2069. Will be fixed if we merge #2124 – have tested it locally and that console warning doesn't appear
